data-prepper-plugins/geoip-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/GeoIPProcessorService.java

```diff
@@ -5,44 +5,137 @@
+import java.util.concurrent.TimeUnit;
 
 /**
  * Implementation class of geoIP-processor plugin service class.
  * It is responsible for calling of mmdb files download
  */
 public class GeoIPProcessorService {
 
+    private static final Logger LOG = LoggerFactory.getLogger(GeoIPProcessorService.class);
+    public static final String DATABASE_1 = "first_database";
+    public static final String DATABASE_2 = "second_database";
+    private GeoIPProcessorConfig geoIPProcessorConfig;
+    private LicenseTypeOptions licenseType;
+    private GetGeoData geoData;
+    private List<DatabasePathURLConfig> databasePath;
+    private final String tempPath;
+    private final ScheduledExecutorService scheduledExecutorService;
+    private final DBSourceOptions dbSourceOptions;
+    public static volatile boolean downloadReady;
+    private boolean toggle;
+    private String flipDatabase;
+
+
     /**
      * GeoIPProcessorService constructor for initialization of required attributes
      * @param geoIPProcessorConfig geoIPProcessorConfig
      * @param tempPath tempPath
      */
     public GeoIPProcessorService(GeoIPProcessorConfig geoIPProcessorConfig, String tempPath) {
-       //TODO
+        this.toggle = false;
+        this.geoIPProcessorConfig = geoIPProcessorConfig;
+        this.tempPath = tempPath;
+        this.databasePath = geoIPProcessorConfig.getServiceType().getMaxMindService().getDatabasePath();
+        flipDatabase = DATABASE_1;
+
+        dbSourceOptions = DbSourceIdentification.getDatabasePathType(databasePath);
+        final Duration checkInterval = Objects.requireNonNull(geoIPProcessorConfig.getServiceType().getMaxMindService().getCacheRefreshSchedule());
+        scheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
+        scheduledExecutorService
+                .scheduleAtFixedRate(this::downloadThroughURLandS3, 0L, checkInterval.toSeconds(), TimeUnit.SECONDS);
+
+        synchronized (this) {
+            try {
+                while (!downloadReady) {
+                    wait();
+                }
+            } catch (InterruptedException ex) {
+                LOG.info("InterruptedException {0} ",  ex);
+                Thread.currentThread().interrupt();
+            }
+            String finalPath = tempPath + File.separator;
+            licenseType = LicenseTypeCheck.isGeoLite2OrEnterpriseLicense(finalPath.concat(flipDatabase));
+            if (licenseType.equals(LicenseTypeOptions.FREE)) {
+                geoData = new GetGeoLite2Data(finalPath.concat(flipDatabase), geoIPProcessorConfig.getServiceType().getMaxMindService().getCacheSize(), geoIPProcessorConfig);
+            }
+            else if (licenseType.equals(LicenseTypeOptions.ENTERPRISE)) {
+                geoData = new GetGeoIP2Data(finalPath.concat(flipDatabase), geoIPProcessorConfig.getServiceType().getMaxMindService().getCacheSize(), geoIPProcessorConfig);
+            }
+        }
+        downloadReady = false;
```

Comment:
I think you should do this before it is used in the "synchronized" block, right?